manifests/init.pp

```diff
@@ -40,30 +40,68 @@
 #   the system will set itself up as a caching nameserver unless this is set
 #   to false.
+# @param manage_via_nmcli
+#   Allows the user to update DNS entries via nmcli instead of directly
+#   modifying resolv.conf
+# @nmcli_device_name
+#   If managing DNS servers via nmcli, this is the device the IPv4 DNS servers
+#   will be added to
+# @auto_reapply_nmcli_device
+#   If true, call nmcli device reapply on the device that had DNS servers added
+#   to it
 # @param sortlist
 #   Optional Array of address/netmask pairs that allow addresses returned by
 #   gethostbyname to be sorted.
 # @param extra_options
 #   Optional Array to put any options that may not be covered by the variables
 #   below. These will be appended to the options string.
 class resolv (
-  Array[Simplib::IP,0,3]     $servers        = simplib::lookup('simp_options::dns::servers', { 'default_value' => ['127.0.0.1'] }),
-  Array[Simplib::Domain,0,6] $search         = simplib::lookup('simp_options::dns::search', { 'default_value'  => [$facts['domain']] }),
-  Resolv::Domain             $resolv_domain  = $facts['domain'],
-  Boolean                    $debug          = false,
-  Boolean                    $rotate         = true,
-  Boolean                    $no_check_names = false,
-  Boolean                    $inet6          = false,
-  Integer[0,15]              $ndots          = 1,
-  Integer[0,30]              $timeout        = 2,
-  Integer[0,5]               $attempts       = 2,
-  Boolean                    $named_server   = false,
-  Boolean                    $named_autoconf = true,
-  Boolean                    $caching        = true,
-  Optional[Resolv::Sortlist] $sortlist       = undef,
-  Optional[Array[String]]    $extra_options  = undef,
+  Array[Simplib::IP,0,3]     $servers                   = simplib::lookup('simp_options::dns::servers', { 'default_value' => ['127.0.0.1'] }),
+  Array[Simplib::Domain,0,6] $search                    = simplib::lookup('simp_options::dns::search', { 'default_value'  => [$facts['domain']] }),
+  Resolv::Domain             $resolv_domain             = $facts['domain'],
+  Boolean                    $debug                     = false,
+  Boolean                    $rotate                    = true,
+  Boolean                    $no_check_names            = false,
+  Boolean                    $inet6                     = false,
+  Integer[0,15]              $ndots                     = 1,
+  Integer[0,30]              $timeout                   = 2,
+  Integer[0,5]               $attempts                  = 2,
+  Boolean                    $named_server              = false,
+  Boolean                    $named_autoconf            = true,
+  Boolean                    $caching                   = true,
+  Boolean                    $manage_via_nmcli          = false,
+  Optional[String]           $nmcli_device_name         = undef,
+  Boolean                    $auto_reapply_nmcli_device = false,
```

Comment:
Not sure why you need this. The DNS settings should be global (I think).

A reapply may trigger unintended updates if applied to a given device where alternative management is in use.

Reply:
Actually, I just figured it out and it's a bit unfortunate.

So, in the near future, we're probably going to run into issues with wanting different DNS servers per interface and needing to hook into `systemd-resolved`.

For now, you probably want to auto-detect the first active nmcli device since this updates global settings anyway.

The `simplib__networkmanager` fact should help with this. That said, if the interface isn't under NetworkManager control I have no idea what will happen in general.